platform/api.progress.nb/src/org/netbeans/api/progress/ProgressHandleFactory.java

```diff
@@ -119,44 +148,171 @@ public static JLabel createMainLabelComponent(ProgressHandle handle) {
      * @param displayName to be shown in the progress UI
      * @return an instance of {@link org.netbeans.api.progress.ProgressHandle}, initialized but not started.
+     * @deprecated Use {@link ProgressHandle#createSystemHandle(java.lang.String, org.openide.util.Cancellable)}.
      */
+    @Deprecated
     public static ProgressHandle createSystemHandle(String displayName) {
         return createSystemHandle(displayName, null, null);
     }
 
     /**
      * Create a cancelable handle for a task that is not triggered by explicit user action.
      * Such tasks have lower priority in the UI.
+     * Since {@code 1.59}, the functionality moves to basic {@link ProgressHandle Progress API}; this method is retained for smooth transition of older API clients.
      * @param displayName to be shown in the progress UI
      * @param allowToCancel either null, if the task cannot be cancelled or 
      *          an instance of {@link org.openide.util.Cancellable} that will be called when user 
      *          triggers cancel of the task.
      * @return an instance of {@link org.netbeans.api.progress.ProgressHandle}, initialized but not started.
+     * @deprecated Use {@link ProgressHandle#createSystemHandle(java.lang.String, org.openide.util.Cancellable)}.
      */
     public static ProgressHandle createSystemHandle(String displayName, Cancellable allowToCancel) {
         return createSystemHandle(displayName, allowToCancel, null);
     }
     
     /**
-     * Create a progress ui handle for a task that is not triggered by explicit user action.
+     * Create a progress UI handle for a task that is not triggered by explicit user action.
+     * Starting from {@code 1.59}, this method is fully superseded by {@link ProgressHandle#createSystemHandle(java.lang.String, org.openide.util.Cancellable, javax.swing.Action)}.
+     * Since {@code 1.59}, the functionality moves to basic {@link ProgressHandle Progress API}; this method is retained for smooth transition of older API clients.
      * @param allowToCancel either null, if the task cannot be cancelled or 
      *          an instance of {@link org.openide.util.Cancellable} that will be called when user 
      *          triggers cancel of the task.
      * @param linkOutput an <code>Action</code> instance that links the running task in the progress bar
      *                   to an output of the task. The action is assumed to open the apropriate component with the task's output.
      * @param displayName to be shown in the progress UI
      * @return an instance of {@link org.netbeans.api.progress.ProgressHandle}, initialized but not started.
-     *
+     * @deprecated Please use {@link ProgressHandle#createSystemHandle(java.lang.String, org.openide.util.Cancellable, javax.swing.Action)}
+     * @Deprecated
      */
     public static ProgressHandle createSystemHandle(String displayName, Cancellable allowToCancel, Action linkOutput) {
```

Comment:
Please prepare a Jackpot hint that would rewrite to `ProgressHandle.createSystemHandle`.